extensions/security/runtime-spi/src/main/java/io/quarkus/security/spi/runtime/AuthorizationEvent.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.security.spi.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.quarkus.security.identity.SecurityIdentity;
+
+/**
+ * Security event that should be fired based on security check result.
+ */
+public final class AuthorizationEvent implements SecurityEvent {
+    public static String AUTHORIZATION_FAILURE_EX_KEY = AuthorizationEvent.class.getName() + "#authZFailureException";
+    public static String CHECKER_CLASS_NAME_KEY = AuthorizationEvent.class.getName() + "#checkerClassName";
+    private final SecurityIdentity securityIdentity;
+    private final Map<String, Object> eventProperties;
+
+    public AuthorizationEvent(SecurityIdentity securityIdentity, Exception authZFailureException,
+            String checkerClassName) {
+        this.securityIdentity = securityIdentity;
+        this.eventProperties = toEventProperties(authZFailureException, checkerClassName);
+
+    }
+
+    public AuthorizationEvent(SecurityIdentity securityIdentity, Exception authZFailureException,
+            String checkerClassName, Map<String, Object> eventProperties) {
+        this(securityIdentity, authZFailureException, checkerClassName);
+        if (eventProperties != null && !eventProperties.isEmpty()) {
+            this.eventProperties.putAll(eventProperties);
+        }
+    }
+
+    public Exception getAuthZFailureException() {
+        return (Exception) eventProperties.get(AUTHORIZATION_FAILURE_EX_KEY);
+    }
+
+    /**
+     * @return name of a class that performed check
+     */
+    public String getCheckerClassName() {
```

Comment:
I'm not sure it can help a lot, may be lets drop it for now and add if someone asks ? IMHO it is worth starting with the minimalistic API

Reply:
But if there is a failure and `RoutingContext` is not available in every single case (thought I did my best to have it available, together with CDI injection it should cover most of it), wouldn't you like to find out what authorization check failed?

That said, the name is bad, I know it. @sberyozkin would you be fine with putting it as event properties map item? also please suggest name, something like `FAILED_CHECK_CLASS_NAME`, not sure. It's valuable information.

Reply:
I mean putting it to map instead of `getCheckerClassName`.

Reply:
@michalvavrik Sure, perhaps indeed, just put that class name as a map property for now if you'd like